test/onnx/dynamo/test_registry_dispatcher.py

```diff
@@ -191,304 +94,330 @@ def setUp(self):
+                    target=torch.ops.aten.add.Tensor,  # type: ignore[attr-defined]
+                    args=(torch.tensor(3), torch.tensor(4)),
+                    kwargs={"alpha": 1},
+                ),
+            ),
+        ]
     )
-    def test_dispatch_opset_version_returns_correct_version(
-        self, available_opsets: Sequence[int], target: int, expected: int
+    def test_respect_custom_ops_in_find_the_perfect_or_nearest_match_onnxfunction(
+        self, node
     ):
-        actual = onnxfunction_dispatcher._dispatch_opset_version(
-            target, available_opsets
+        custom_domain = onnxscript.values.Opset(domain="custom", version=1)
+
+        @onnxscript.script(custom_domain)
+        def test_custom_op(x: TCustomFloat, y: TCustomFloat) -> TCustomFloat:
+            return op.Add(x, y)
+
+        @onnxscript.script(custom_domain)
+        def test_default_op(x: TCustomFloat, y: TCustomFloat) -> TCustomFloat:
+            return op.Add(x, y)
+
+        custom_overloads = set(
+            {
+                registration.SymbolicFunction(
+                    test_custom_op, op_name="test::test_op", is_custom=True
+                )
+            }
+        )
+        function_overloads = (
+            set(
+                {
+                    registration.SymbolicFunction(
+                        test_default_op, op_name="test::test_op"
+                    )
+                }
+            )
+            | custom_overloads
         )
-        self.assertEqual(actual, expected)
-
 
-class TestOpSchemaWrapper(common_utils.TestCase):
-    def setUp(self):
-        # OnnxFunction with default attributes
-        self.onnx_function_add = ops.core.aten_add
-        self.op_schema_wrapper_add = onnxfunction_dispatcher._OpSchemaWrapper(
-            self.onnx_function_add.op_schema
+        symbolic_fn = self.dispatcher._find_the_perfect_or_nearest_match_onnxfunction(
+            node,
+            "aten",
+            "add",
+            None,
+            function_overloads,
+            custom_overloads,  # custom function overloads
+            node.args,
+            node.kwargs,
+            self.diagnostic_context,
         )
+        self.assertEqual(symbolic_fn, test_custom_op)
 
-        # overload type: overloaded dtypes
-        self.onnx_function_mul = ops.core.aten_mul
-        self.onnx_function_mul_bool = ops.core.aten_mul_bool
 
+class TestOpSchemaWrapper(common_utils.TestCase):
+    def setUp(self):
         # overload type: optional dtype
         self.onnx_function_new_full = ops.core.aten_new_full
         self.onnx_function_new_full_dtype = ops.core.aten_new_full_dtype
 
-    def test_perfect_match_inputs(self):
-        inputs = [torch.randn(3, 4), torch.randn(3, 4)]
-        wrong_type_inputs = ["A", "B"]
-        wrong_number_of_inputs = [
-            torch.randn(3, 4),
-            torch.randn(3, 4),
-            torch.randn(3, 4),
+    @parameterized.expand(
```

Comment:
Just a note to myself that I am here. Will continue the review soon

Reply:
I need to see it. Thanks lol. I will resolve myself

Reply:
Oh sorry, I thought this was for 3 hours ago of you.